c7n/resources/redshift.py

```diff
@@ -601,6 +601,57 @@ def process(self, clusters):
                 VpcSecurityGroupIds=groups[idx])
 
 
+
+@RedshiftSnapshot.action_registry.register('copy-cluster-tags')
+class CopyClusterTags(BaseAction):
+
+    """
+    Copy specific tags from cluster to Snapshot
+    :example:
+
+        .. code-block: yaml
+
+            - name: redshiftsnapshot-copytags
+              resource: redshift-snapshot
+              actions:
+                - type: copy-cluster-tags
+                  tags:
+                    - required-tag
+                    - required-tag1
+    """
+
+    schema = type_schema(
+        'copy-cluster-tags',
+        tags={'type': 'array', 'items': {'type': 'string'}, 'minItems': 1},
+        required = ('tags',))
+
+    def get_permissions(self):
+        perms = self.manager.get_resource_manager('redshift').get_permissions()
+        perms.append('redshift:CreateTags',)
+        return perms
+
+    def process(self, snapshots):
+        log.info("Modifying %d redshift snapshots", len(snapshots))
+        client = local_session(self.manager.session_factory).client('redshift')
+        clusters = {
+            cluster['ClusterIdentifier']: cluster for cluster in
+            self.manager.get_resource_manager('redshift').resources()}
+
+        for s in snapshots:
+            if s['ClusterIdentifier'] in clusters:
```

Comment:
I'd restructure this to avoid indenting what follows ie. if s['ClusterIdentifer'] not in clusters: continue

and then dedent the remainder.